### ISSUE TITLE
extract_relevant_features() doesn't work if y isn't sorted by index

### ISSUE THREAD

**Issue by PavloFesenko** [blue-yonder/tsfresh#713]
1. Linux 3.10.0-1062.9.1.el7.x86_64
2. tsfresh 0.16.0
3. Sorry, I can't post the data here due to privacy reasons but I will try to describe the problem as well as I can. I have a target for a regression problem but the function `extract_relevant_features()` doesn't return any relevant features. However, if I sort my Pandas DataFrame by "id" before splitting it in "timeseries" and "y", then everything works. I think the problem might be in the function `select_features()` because if I do step-by-step `extract_features()`, `impute()` and `select_features()` without sorting "id", then the relevant features aren't returned as well. By the way, I couldn't reproduce this issue for the robot failures dataset which is a classification problem so it might be only relevant for regression. If there is maybe a toy dataset for regression, then I could provide a minimal code snippet. Thanks a lot!

**Comment by nils-braun** [comment]
Hi @PavloFesenko! 
Thank you very much for your issue! 
I am not quite sure I can understand the issue, but it might look like some indexing problems. What are the types of data (I guess a dataframe for `df` and a pandas series for `y` or only a NumPy array)?

Here is some code to generate a test regression data sample:
```python
from tsfresh import extract_relevant_features
from tsfresh.examples import load_driftbif

df, y = load_driftbif(100, 10, classification=False, seed=42)

print(df.head())
print(y.head())

X = extract_relevant_features(df, y, column_id="id", column_sort="time", column_kind="dimension", column_value="value")
print(X.head())
```
At least for this data sample, I get some extracted features (360 in this case).
Can you somehow map this to your problem?

**Comment by PavloFesenko** [comment]
Hi @nils-braun!

Thank you for your prompt response! 😊

The issue appears when the column "id" is a string instead of an integer (as is the case in my dataset). I added two lines that change the type of the column "id" and the type of the index for `y` to string. If you try the following code, you would only see 1 extracted feature:

``` python
from tsfresh import extract_relevant_features
from tsfresh.examples import load_driftbif

df, y = load_driftbif(100, 10, classification=False, seed=42)

df['id'] = df['id'].astype('str')
y.index = y.index.astype('str')

X = extract_relevant_features(df, y, column_id="id", column_sort="time", column_kind="dimension", column_value="value")
print(X.head())
```

If you, however, change the task to `classification=True`, you will get 360 features.

Hope this will help to debug the issue. 😉

**Comment by nils-braun** [comment]
Hi @PavloFesenko,

yes, that definitely helped! (and also, please feel free free to change the title :-))

I could reproduce the problem and I could also find a quickfix. If you want, could you try out?

Thank you very much for reporting this. I am not 100% sure already, but I think this is quite a large bug :-/

**Comment by nils-braun** [comment]
@PavloFesenko - I think the issue was solved in #715 (at least for your MWE).
I am closing this ticket now - but feel free to re-open if the PR did not help! :-)
